core/broker/broker.go

```diff
@@ -134,10 +135,8 @@ func (b *broker) Init(c *component.Component) error {
 	b.checkPrefixAnnouncements()
 	b.Component.SetStatus(component.StatusHealthy)
 	if b.Component.Monitor != nil {
-		b.monitorStream = b.Component.Monitor.NewBrokerStreams(b.Identity.Id, b.AccessToken)
-		go b.Component.Monitor.TickStatus(func() {
-			b.monitorStream.Send(b.GetStatus())
-		})
+		b.monitorStream = b.Component.Monitor.BrokerClient(b.Context, grpc.PerRPCCredentials(auth.WithStaticToken(b.AccessToken)))
+		go b.Component.Monitor.TickStatus(func() { b.monitorStream.Send(b.GetStatus()) })
```

Comment:
this goroutine never returns. Which may cause trouble. What if the component is down/disconnected? Do we need to query it's status all the time and send it?